emu/src/alu_instruction.rs

```diff
@@ -17,6 +17,26 @@ pub enum ArmModeAluInstruction {
     Mvn = 0xF,
 }
 
+#[derive(Eq, PartialEq)]
+pub enum AluInstructionTypes {
+    Logical,
+    Arithmetic,
+}
+
+pub trait Type {
+    fn get_type(&self) -> AluInstructionTypes;
+}
+
+impl Type for ArmModeAluInstruction {
+    fn get_type(&self) -> AluInstructionTypes {
```

Comment:
Better
```suggestion
    fn type(&self) -> AluInstructionTypes {
```
or
```suggestion
    fn kind(&self) -> AluInstructionTypes {
```

Reply:
I like `type` more but it seems like you can't have a function called `type` in rust.
I don't like `kind` that much since I think the term type is more appropriate in this case. What do you think?

Reply:
Ok for me :)